### ISSUE TITLE
Cognito and creating user pool client fails on first try with CDK, but succeeds the second time.  Any ideas?

### ISSUE THREAD

**Issue by BStenfors9119** [localstack/localstack#5859]
### Is there an existing issue for this?

- [X] I have searched the existing issues and read the documentation

### Question

Hello, I've been using localstack pro for a bit and love it.  I am running into an issue with Cognito and creating a user pool client on the first try with CDK.  Here is the error(s) I am seeing.  If I run the same cdk app a second time it succeeds:

`docker_localstack_1 | 2022-04-11T22:12:34.546:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for Cloud Formation stack "cfd-inspector-idp-local": 'NoneType' object has no attribute 'get' Traceback (most recent call last):
docker_localstack_1 |  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1531, in _run
docker_localstack_1 |   self.do_apply_changes_in_loop(changes, stack, stack_name)
docker_localstack_1 |  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1582, in do_apply_changes_in_loop`

Here is what the CDK (in goLang) of the user pool client creation looks like:

`samlIdp := &CfdIdp{
		IdpName: jsii.String(fmt.Sprintf("%s", envName)),
	}`

	`userPoolClient := userPool.AddClient(jsii.String("user-pool-client"), &awscognito.UserPoolClientOptions{
		AccessTokenValidity: awscdk.Duration_Hours(jsii.Number(1)),
		AuthFlows:           &awscognito.AuthFlow{Custom: jsii.Bool(true), UserSrp: jsii.Bool(true)},
		GenerateSecret:      jsii.Bool(true),
		IdTokenValidity:     awscdk.Duration_Hours(jsii.Number(1)),
		OAuth: &awscognito.OAuthSettings{
			Flows:        &awscognito.OAuthFlows{ImplicitCodeGrant: jsii.Bool(true)},
			Scopes:       &[]awscognito.OAuthScope{awscognito.OAuthScope_OPENID(), awscognito.OAuthScope_EMAIL(), awscognito.OAuthScope_PROFILE()},
			CallbackUrls: &[]*string{redirectUri},
			LogoutUrls:   &[]*string{jsii.String(logoutUrl)},
		},
		PreventUserExistenceErrors: jsii.Bool(true),
		RefreshTokenValidity:       awscdk.Duration_Days(jsii.Number(30)),
		SupportedIdentityProviders: &[]awscognito.UserPoolClientIdentityProvider{samlIdp},
		UserPoolClientName:         jsii.String("web-app-client"),
	})`

**Comment by whummer** [comment]
Thanks for reporting @BStenfors9119 . Can you please share the full logs, including the full stack trace of the error you're seeing? This would be very helpful for us to track down the problem. Also, would you mind sharing the CloudFormation template JSON file that gets generated in your `cdk.out` folder, to help us reproduce this issue? Thanks for your help!

**Comment by BStenfors9119** [comment]
Full stack trace:
`2022-04-17T22:38:13.576:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for Cloud 
Formation stack "cfd-inspector-idp-local": 'NoneType' object has no attribute 'get' Traceback (most recent call last):                  
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1531, in _run            
docker_localstack_1 |     self.do_apply_changes_in_loop(changes, stack, stack_name)                                                     
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1582, in do_apply_change 
s_in_loop                                                                                                                               
docker_localstack_1 |     should_deploy = self.prepare_should_deploy_change(                                                            
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1641, in prepare_should_ 
deploy_change                                                                                                                           
docker_localstack_1 |     resolve_refs_recursively(stack, resource)                                                                     
docker_localstack_1 |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/cloudformation/cloudformat 
ion_extended.py.enc", line 321, in resolve_refs_recursively                                                                             
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                       
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                               
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 483, in resolve_refs_rec 
ursively                                                                                                                                
docker_localstack_1 |     result = _resolve_refs_recursively(stack, value)                                                              
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                       
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                               
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 679, in _resolve_refs_re 
cursively                                                                                                                               
docker_localstack_1 |     value[key] = resolve_refs_recursively(stack, val)                                                             
docker_localstack_1 |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/cloudformation/cloudformat
ion_extended.py.enc", line 321, in resolve_refs_recursively                                                                             
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                       
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                               
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 483, in resolve_refs_rec
ursively                                                                                                                                
docker_localstack_1 |     result = _resolve_refs_recursively(stack, value)                                                              
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                       
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                               
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 679, in _resolve_refs_re
cursively                                                                                                                              
docker_localstack_1 |     value[key] = resolve_refs_recursively(stack, val)                                                             
docker_localstack_1 |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/cloudformation/cloudformat
ion_extended.py.enc", line 321, in resolve_refs_recursively                                                                            
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                      
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                              
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 483, in resolve_refs_rec
ursively                                                                                                                               
docker_localstack_1 |     result = _resolve_refs_recursively(stack, value)                                                             
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                      
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                              
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 574, in _resolve_refs_re 
cursively                                                                                                                              
docker_localstack_1 |     join_values = [resolve_refs_recursively(stack, v) for v in join_values]                                      
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 574, in <listcomp>       
docker_localstack_1 |     join_values = [resolve_refs_recursively(stack, v) for v in join_values]                                      
docker_localstack_1 |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/cloudformation/cloudformat
ion_extended.py.enc", line 321, in resolve_refs_recursively                                                                            
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                      
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                              
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 483, in resolve_refs_rec
ursively                                                                                                                               
docker_localstack_1 |     result = _resolve_refs_recursively(stack, value)                                                             
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/common.py", line 661, in func                                      
docker_localstack_1 |     return wrapped(*args, **kwargs)                                                                               
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 558, in _resolve_refs_re
cursively                                                                                                                              
docker_localstack_1 |     ref = resolve_ref(stack, value["Ref"], attribute="Ref")                                                      
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 444, in resolve_ref     
docker_localstack_1 |     return determine_resource_physical_id(                                                                       
docker_localstack_1 |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1091, in determine_resou
rce_physical_id                                                                                                                        
docker_localstack_1 |     result = extract_resource_attribute(                                                                         
docker_localstack_1 |   File "/opt/code/localstack/.venv/lib/python3.8/site-packages/localstack_ext/services/cloudformation/cloudformat
ion_extended.py.enc", line 364, in extract_resource_attribute                                                                          
docker_localstack_1 | AttributeError: 'NoneType' object has no attribute 'get'`


[cdkout.zip](https://github.com/localstack/localstack/files/8502576/cdkout.zip)

Where / how do I obtain the "full logs"?

TIA

**Comment by whummer** [comment]
Hi @BStenfors9119 , thanks for reporting. We have now pushed a few enhancements which should (hopefully) resolve the issue you've been facing. Can you please pull the `latest` Docker image and give it another try? Please keep us posted on how it goes.. Thanks!

**Comment by BStenfors9119** [comment]
@whummer Pulled the latest and it works!

Thank you so much! This allows me to kick off my tmux "app" now and just let the system set itself up without having to watch for the crash to rerun it :).

Thank you again!

**Comment by whummer** [comment]
Hi @BStenfors9119 , thanks for the update. Can you please share the 

```
Auth parameters username/password not specified
```

I just gave this a try with the `latest` Docker image - here are the steps to reproduce:
```
$ pool_id=$(awslocal cognito-idp create-user-pool --pool-name pool1 | jq -r .UserPool.Id)
$ client_id=$(awslocal cognito-idp create-user-pool-client --user-pool-id $pool_id --client-name c1 | jq -r .UserPoolClient.ClientId)
$ open "http://localhost:4566/login?response_type=code&redirect_uri=http://example.com&client_id=$client_id"
```

Then, in the Cognito browser window: switching to "Signup", creating a user: 
<img width="324" alt="image" src="https://user-images.githubusercontent.com/2807888/168430759-42fed5d6-e364-48bf-8aff-c4c26a5b6280.png">

Once created, we can use the CLI to confirm the user:
```
$ awslocal cognito-idp admin-confirm-sign-up --user-pool-id $pool_id --username test
```

Then using the UI to sign in:
<img width="353" alt="image" src="https://user-images.githubusercontent.com/2807888/168430798-1bdb8dba-6a3a-49e8-b4a4-99a5a2d1e1ba.png">

... which ultimately redirects the browser to the redirect URI (http://example.com) including the `code` parameter:
<img width="225" alt="image" src="https://user-images.githubusercontent.com/2807888/168430845-705701e1-33a5-4abe-be75-20a4286570cd.png">

Are you able to replicate these basic steps with the `latest` Docker image? Thanks for your help in resolving this!

**Comment by BStenfors9119** [comment]
@whummer thank you for the reply and work.  

I apologize for the delayed response but just now getting back to this project.
I can confirm as of right now this issue seems to be fixed using version 0.14.3.1.

Again thank you for the hard work and responses!
